lib/AST/RequirementMachine/RequirementLowering.cpp

```diff
@@ -596,6 +598,13 @@ struct InferRequirementsWalker : public TypeWalker {
         constrainParametersAndResult(fnTy->getDifferentiabilityKind() ==
                                      DifferentiabilityKind::Linear);
       }
+
+      // Infer that the thrown error type conforms to Error.
+      if (auto thrownError = fnTy->getThrownError()) {
```

Comment:
Maybe mention here that this is for the `throws(E)` inside of a function type

Reply:
I've done this in https://github.com/apple/swift/pull/69198